OpenDreamRuntime/Procs/DMProc.cs

```diff
@@ -37,104 +37,99 @@ public sealed class DMProcState : ProcState
-            DMOpcodeHandlers.CompareLessThanOrEqual,
-            DMOpcodeHandlers.CreateAssociativeList,
-            DMOpcodeHandlers.Remove,
-            DMOpcodeHandlers.DeleteObject,
-            DMOpcodeHandlers.PushResource,
-            DMOpcodeHandlers.CreateList,
-            DMOpcodeHandlers.CallStatement,
-            DMOpcodeHandlers.BitAnd,
-            DMOpcodeHandlers.CompareNotEquals,
-            DMOpcodeHandlers.Divide,
-            DMOpcodeHandlers.Multiply,
-            DMOpcodeHandlers.BitXorReference,
-            DMOpcodeHandlers.BitXor,
-            DMOpcodeHandlers.BitOr,
-            DMOpcodeHandlers.BitNot,
-            DMOpcodeHandlers.Combine,
-            DMOpcodeHandlers.CreateObject,
-            DMOpcodeHandlers.BooleanOr,
-            DMOpcodeHandlers.PushArgumentList,
-            DMOpcodeHandlers.CompareGreaterThanOrEqual,
-            DMOpcodeHandlers.SwitchCase,
-            DMOpcodeHandlers.Mask,
-            DMOpcodeHandlers.Error,
-            DMOpcodeHandlers.IsInList,
-            DMOpcodeHandlers.PushArguments,
-            DMOpcodeHandlers.PushFloat,
-            DMOpcodeHandlers.ModulusReference,
-            DMOpcodeHandlers.CreateListEnumerator,
-            DMOpcodeHandlers.Enumerate,
-            DMOpcodeHandlers.DestroyEnumerator,
-            DMOpcodeHandlers.Browse,
-            DMOpcodeHandlers.BrowseResource,
-            DMOpcodeHandlers.OutputControl,
-            DMOpcodeHandlers.BitShiftRight,
-            null, //0x41
-            DMOpcodeHandlers.Power,
-            DMOpcodeHandlers.DebugSource,
-            DMOpcodeHandlers.DebugLine,
-            DMOpcodeHandlers.Prompt,
-            DMOpcodeHandlers.PushProcArguments,
-            DMOpcodeHandlers.Initial,
-            null, //0x48
-            DMOpcodeHandlers.IsType,
-            DMOpcodeHandlers.LocateCoord,
-            DMOpcodeHandlers.Locate,
-            DMOpcodeHandlers.IsNull,
-            DMOpcodeHandlers.Spawn,
-            DMOpcodeHandlers.OutputReference,
-            DMOpcodeHandlers.Output,
-            DMOpcodeHandlers.JumpIfNullDereference,
-            DMOpcodeHandlers.Pop,
-            DMOpcodeHandlers.Prob,
-            DMOpcodeHandlers.IsSaved,
-            DMOpcodeHandlers.PickUnweighted,
-            DMOpcodeHandlers.PickWeighted,
-            DMOpcodeHandlers.Increment,
-            DMOpcodeHandlers.Decrement,
-            DMOpcodeHandlers.CompareEquivalent,
-            DMOpcodeHandlers.CompareNotEquivalent,
-            DMOpcodeHandlers.Throw,
-            DMOpcodeHandlers.IsInRange,
-            DMOpcodeHandlers.MassConcatenation,
-            DMOpcodeHandlers.CreateTypeEnumerator,
-            null, //0x5E
-            DMOpcodeHandlers.PushGlobalVars,
-            DMOpcodeHandlers.ModulusModulus,
-            DMOpcodeHandlers.ModulusModulusReference
+        private static readonly Dictionary<DreamProcOpcode, OpcodeHandler?> _opcodeHandlers = new Dictionary<DreamProcOpcode, OpcodeHandler?>(){
```

Comment:
Possibly? It depends how smart C# is at optimising under-the-hood. I'll investigate. 

I did primarily do this so that we wouldn't have floating unconnected opcode stuff anymore, and you don't need to carefully maintain two lists when modifying them.

Reply:
It has just occurred to me that I could pretty trivially use this structure for maintainability, and then have an init proc that just turns into a nice lookup array so you can get best of both worlds.

Reply:
That could be a decent solution